### ISSUE TITLE
Tag categories and the relations tables

### ISSUE THREAD

**Issue by JMurphyWeb** [postgres-plugin/tags-system#35]
We need to introduce a tag-categories table as well as a tags-categories-relations table.

We also need a name for the 'linking' table.

### Current thoughts

tag_categories:
```sql
CREATE TABLE IF NOT EXISTS tag_categories
  (id INTEGER PRIMARY KEY,
  name VARCHAR(50) NOT NULL,
  active BOOLEAN NOT NULL);

TRUNCATE tag_categories;
```

tags_categories_relations:
```sql
CREATE TABLE IF NOT EXISTS tags_categories_relations
  (category_id INTEGER REFERENCES tag_categories (id),
  tag_id INTEGER REFERENCES tags (id),
  active BOOLEAN NOT NULL);

TRUNCATE tags_categories_relations;

```

**Comment by JMurphyWeb** [comment]
Turns out `-` is not permitted in table names, replacing with `_`

**Comment by JMurphyWeb** [comment]
tested in the client, all added fine.  Next I need to try inserting some entries.

**Comment by JMurphyWeb** [comment]
I actually got an error when trying to truncate the table which was referenced as a `FOREIGN KEY`

`error: cannot truncate a table referenced in a foreign key constraint`

**Comment by JMurphyWeb** [comment]
```sql
TRUNCATE tags CASCADE;
```

Adding `CASCADE` to the truncate seemed to fix

**Comment by JMurphyWeb** [comment]
SQL file is now looking like this:

```sql
-------------------------------
---------    TAGS    ----------

-- Check if 'tags' table exists already; create if not
CREATE TABLE IF NOT EXISTS tags (
  id INTEGER PRIMARY KEY,
  name VARCHAR(50) NOT NULL,
  active BOOLEAN NOT NULL
);

-- Delete all entries of the tags table
TRUNCATE tags CASCADE;

-------------------------------
----    TAG_CATEGORIES    -----

CREATE TABLE IF NOT EXISTS tag_categories
  (id INTEGER PRIMARY KEY,
  name VARCHAR(50) NOT NULL,
  active BOOLEAN NOT NULL);

TRUNCATE tag_categories CASCADE;


-------------------------------
-- TAGS_CATEGORIES_RELATIONS --

CREATE TABLE IF NOT EXISTS tags_categories_relations
  (category_id INTEGER REFERENCES tag_categories (id),
  tag_id INTEGER REFERENCES tags (id),
  active BOOLEAN NOT NULL);

TRUNCATE tags_categories_relations CASCADE;

```

**Comment by SimonLab** [comment]
resolved with #41
